src/entt/entity/mixin.hpp

```diff
@@ -89,9 +91,10 @@ class sigh_mixin final: public Type {
     /**
      * @brief Constructs an empty storage with a given allocator.
      * @param allocator The allocator to use.
+     * @param traits The type_traits to use.
      */
-    explicit sigh_mixin(const allocator_type &allocator)
-        : Type{allocator},
+    explicit sigh_mixin(const allocator_type &allocator, traits_type traits = {})
```

Comment:
The reason why the trait is made staeful in this PR is that we need an object that can be specialized for the component type. And the trait itself (if we ignore its naming) fits the role perfectly because it effectively specialized the storage based on the component type already (page size, empty type). Why not use it for any other use cases where the storage needs to be specialized?

Reply:
I think _because it leaks to classes that don't care about it at all_ is my first objection to _using it for any other use cases where the storage need to be specialized_.
In terms of software design, I generally look suspiciously at my code if it leaks private information from a base class to a derived class. This is exactly the case from what I get. For example, the `sigh_mixin` as well as any other mixin doesn't care about these traits.
Also, does it mean that all existing mixin out there must be updated accordingly? Because I've a bunch of them in production code that I can't easily touch/change/update.